lib/matplotlib/tests/test_cbook.py

```diff
@@ -651,3 +651,41 @@ def test_check_shape(target, test_shape):
     with pytest.raises(ValueError,
                        match=error_pattern):
         cbook._check_shape(target, aardvark=data)
+
+
+def test_setattr_cm():
+    class A:
+        def __init__(self):
+            self.aardvark = 'aardvark'
+            self._p = 'p'
+
+        def meth(self):
+            ...
+
+        @property
+        def prop(self):
+            return self._p
+
+        @prop.setter
+        def prop(self, val):
+            self._p = val
+
+    a = A()
+
+    assert id(a.meth) != id(a.meth)
```

Comment:
I know it's because of descriptors or whatever, but I feel that this needs a comment.